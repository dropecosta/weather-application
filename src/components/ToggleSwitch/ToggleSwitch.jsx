import React from 'react';
import './ToggleSwitch.scss';

const ToggleSwitch = ({ isOn, handleToggle, onColor }) => {
  return (
    <>
      <input
        checked={isOn}
        onChange={handleToggle}
        className="switch-checkbox"
        id={`switch-new`}
        type="checkbox"
      />
      <label
        style={{ background: isOn && onColor }}
        className="switch-label"
        htmlFor={`switch-new`}
      >
        <span className={`switch-button`} />
      </label>
    </>
  );
};

export default ToggleSwitch
