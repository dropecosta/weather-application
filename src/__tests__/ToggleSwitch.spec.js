import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import ToggleSwitch from "../components/ToggleSwitch/ToggleSwitch";

describe("ToggleSwitch component tests", () => {
  it("should render APP component with container class", () => {
    const { container } = render(<ToggleSwitch />);
    expect(container.firstChild).toHaveClass("switch-checkbox");
  });

  it("should match the snapshot", () => {
    const tree = renderer.create(<ToggleSwitch />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
