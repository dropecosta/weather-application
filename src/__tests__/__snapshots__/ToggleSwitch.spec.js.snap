// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch component tests should match the snapshot 1`] = `
Array [
  <input
    className="switch-checkbox"
    id="switch-new"
    type="checkbox"
  />,
  <label
    className="switch-label"
    htmlFor="switch-new"
    style={
      Object {
        "background": undefined,
      }
    }
  >
    <span
      className="switch-button"
    />
  </label>,
]
`;
